packages/ai-ide/src/browser/task-background-summary-variable.ts

```diff
@@ -0,0 +1,67 @@
+// *****************************************************************************
+// Copyright (C) 2025 EclipseSource GmbH and others.
+//
+// This program and the accompanying materials are made available under the
+// terms of the Eclipse Public License v. 2.0 which is available at
+// http://www.eclipse.org/legal/epl-2.0.
+//
+// This Source Code may also be made available under the following Secondary
+// Licenses when the conditions for such availability set forth in the Eclipse
+// Public License v. 2.0 are satisfied: GNU General Public License, version 2
+// with the GNU Classpath Exception which is available at
+// https://www.gnu.org/software/classpath/license.html.
+//
+// SPDX-License-Identifier: EPL-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0
+// *****************************************************************************
+
+import { MaybePromise, nls } from '@theia/core';
+import { injectable } from '@theia/core/shared/inversify';
+import {
+    AIVariable,
+    ResolvedAIVariable,
+    AIVariableContribution,
+    AIVariableService,
+    AIVariableResolutionRequest,
+    AIVariableContext,
+    AIVariableResolverWithVariableDependencies,
+    AIVariableArg
+} from '@theia/ai-core';
+import { ChatSessionContext } from '@theia/ai-chat';
+import { TASK_CONTEXT_VARIABLE } from '@theia/ai-chat/lib/browser/task-context-variable-contribution';
+import { TASK_CONTEXT_SUMMARY_VARIABLE_ID } from '../common/context-variables';
+
+export const TASK_CONTEXT_SUMMARY_VARIABLE: AIVariable = {
+    id: TASK_CONTEXT_SUMMARY_VARIABLE_ID,
+    description: nls.localize('theia/ai/core/taskContextSummary/description', 'Resolves all task context items present in the session context.'),
+    name: TASK_CONTEXT_SUMMARY_VARIABLE_ID,
+};
+
+@injectable()
+/**
+ * @class provides a summary of all TaskContextVariables in the context of a given session. Oriented towards use in prompts.
+ */
+export class TaskContextSummaryVariableContribution implements AIVariableContribution, AIVariableResolverWithVariableDependencies {
+    registerVariables(service: AIVariableService): void {
+        service.registerResolver(TASK_CONTEXT_SUMMARY_VARIABLE, this);
+    }
+
+    canResolve(request: AIVariableResolutionRequest, context: AIVariableContext): MaybePromise<number> {
+        return request.variable.name === TASK_CONTEXT_SUMMARY_VARIABLE.name ? 50 : 0;
+    }
+
+    async resolve(
```

Comment:
If there is no summary, this still resolves to {{taskContextSummary}}, we should explicitly resolve to empty String like done here: packages/ai-core/src/common/prompt-variable-contribution.ts

Reply:
Hm... given that `undefined` is a legitimate return value according to the interface, it seems to me that it ought to be treated the same as an `''` for the purposes of printing the prompt - is there any circumstance in which printing the `{{}}` template value in place makes sense when the return is `undefined`?

Reply:
Yes, for variables, where we allow "empty" I would return ''. For variables which should be resolved (and by default) we return {{}} to ease debugging (you can see that the variable has not been resolved. This is maybe not perfect, but this is how I would go for now (see the example I posted where also "empty" is a valid value.